src/claimant.js

```diff
@@ -0,0 +1,171 @@
+ *
+ * The claim predicate is optional, it defaults to unconditional if none is specified.
+ *
+ * @constructor
+ * @param {string} destination - The destination account ID.
+ * @param {xdr.ClaimPredicate} [predicate] - The claim predicate.
+ */
+export class Claimant {
+  constructor(destination, predicate) {
+    if (destination && !StrKey.isValidEd25519PublicKey(destination)) {
+      throw new Error('Destination is invalid');
+    }
+    this._destination = destination;
+
+    if (!predicate) {
+      this._predicate = xdr.ClaimPredicate.claimPredicateUnconditional();
+    } else if (predicate instanceof xdr.ClaimPredicate) {
+      this._predicate = predicate;
+    } else {
+      throw new Error('Predicate should be an xdr.ClaimPredicate');
+    }
+  }
+
+  /**
+   * Returns an unconditional claim predicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateUnconditional() {
+    return xdr.ClaimPredicate.claimPredicateUnconditional();
+  }
+
+  /**
+   * Returns an `and` claim predicate
+   * @param {xdr.ClaimPredicate} left an xdr.ClaimPredicate
+   * @param {xdr.ClaimPredicate} right an xdr.ClaimPredicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateAnd(left, right) {
+    if (!(left instanceof xdr.ClaimPredicate)) {
+      throw new Error('left Predicate should be an xdr.ClaimPredicate');
+    }
+    if (!(right instanceof xdr.ClaimPredicate)) {
+      throw new Error('right Predicate should be an xdr.ClaimPredicate');
+    }
+
+    return xdr.ClaimPredicate.claimPredicateAnd([left, right]);
+  }
+
+  /**
+   * Returns an `or` claim predicate
+   * @param {xdr.ClaimPredicate} left an xdr.ClaimPredicate
+   * @param {xdr.ClaimPredicate} right an xdr.ClaimPredicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateOr(left, right) {
+    if (!(left instanceof xdr.ClaimPredicate)) {
+      throw new Error('left Predicate should be an xdr.ClaimPredicate');
+    }
+    if (!(right instanceof xdr.ClaimPredicate)) {
+      throw new Error('right Predicate should be an xdr.ClaimPredicate');
+    }
+
+    return xdr.ClaimPredicate.claimPredicateOr([left, right]);
+  }
+
+  /**
+   * Returns a `not` claim predicate
+   * @param {xdr.ClaimPredicate} predicate an xdr.ClaimPredicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateNot(predicate) {
+    if (!(predicate instanceof xdr.ClaimPredicate)) {
+      throw new Error('right Predicate should be an xdr.ClaimPredicate');
+    }
+
+    return xdr.ClaimPredicate.claimPredicateNot(predicate);
+  }
+
+  /**
+   * Returns a `BeforeAbsoluteTime` claim predicate
+   *
+   * This predicate will be fulfilled if the closing time of the ledger that
+   * includes the CreateClaimableBalance operation is less than this (absolute) Unix timestamp.
+   *
+   * @param {string} absBefore Unix epoch as a string
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateBeforeAbsoluteTime(absBefore) {
```

Comment:
Any reason these are strings rather than integers?

Reply:
@Shaptic this can be an int64 value which JS doesn't support properly. So the workaround is to take the value as a String  - and then let the application calculate this value (they can use libraries like https://www.npmjs.com/package/long)